### PR TITLE
chore: deprecated sdk methods

### PR DESCRIPTION
**Pull request by rb-determined-ai**
The following methods were deprecated in 0.17.9 (Feb 2022):
- Checkpoint.load()
- Checkpoint.load_from_path()
- Checkpoint.parse_metadata()
- Checkpoint.get_type()
- Checkpoint.from_json()

They are now removed.

### CODE REVIEW

File: webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx [hunk 1]
```diff
@@ -188,6 +188,7 @@ ckpt = version.checkpoint
 ################ Approach 1 ################
 # You can load the trial directly without having to instantiate the model.
 # The trial should have the model as an attribute.
+# XXX: can we link to the docs instead of embedding docs here?
```
Comment (rb-determined-ai) [inline webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx:191]: @EmilyBonar this question is for you I think... what is the best strategy here?

**Review by azhou-determined — APPROVED** [review]
nice

File: harness/determined/common/experimental/checkpoint/_checkpoint.py [hunk 2]
```diff
@@ -432,16 +319,6 @@ def _parse_metadata(directory: pathlib.Path) -> Dict[str, Any]:
...
-        warnings.warn(
-            "Checkpoint.parse_metadata() is deprecated and will be removed from the public API "
-            "in a future version",
-            FutureWarning,
-            stacklevel=2,
-        )
-        return Checkpoint._parse_metadata(directory)
-
     @staticmethod
     def _get_type(metadata: Dict[str, Any]) -> ModelFramework:
```
Comment (azhou-determined) [inline harness/determined/common/experimental/checkpoint/_checkpoint.py:323]: this too?

**Review by keita-determined — APPROVED** [review]
LGTM
i think we want code highlight here, i'll make a ticket
image

File: docs/release-notes/6268-ckpt.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+:orphan:
+
+**Removed Features**
+
+-  The following methods of :class:`~determined.experimental.client.Checkpoint`,
+   :class:`~determined.experimental.client.Model`, and
+   :class:`~determined.experimental.client.ModelVersion` were deprecated in
+   0.17.9 (Feb 2022) and have are now removed:
```
Comment (tara-det-ai) [inline docs/release-notes/6268-ckpt.txt:8]: @rb-determined-ai 

` 0.17.9 (Feb 2022) and are now removed:`

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Pull request merged**